playbooks/common/roles/common/tasks/main.yaml

```diff
@@ -30,13 +30,26 @@
 
 # Make sure we have private IPs in /etc/hosts
 # Taken from https://gist.github.com/rothgar/8793800
-- name: "Build hosts file"
+- name: "Build hosts file (using private network IPs)"
   lineinfile:
     dest: /etc/hosts
     regexp: ".*{{ item }}$"
     line: "{{ hostvars[item].private_ip }} {{ item }}"
     state: present
-  when: "hostvars[item].private_ip is defined"
+  when:
+    - hostvars[item].private_ip is defined
+    - use_public_network is not defined or use_public_network != True
+  with_items: "{{ groups['satellite6'] + groups['capsules'] }}"
+
+- name: "Build hosts file (using public network IPs)"
+  lineinfile:
+    dest: /etc/hosts
+    regexp: ".*{{ item }}$"
+    line: "{{ hostvars[item].public_ip }} {{ item }}"
+    state: present
+  when:
+    - hostvars[item].public_ip is defined
+    - use_public_network is defined and use_public_network == True
```

Comment:
I'm bit concerned about the volume of variables we have already. What about:
```
- name: "Build hosts file (using private network IPs)"
    ...
  when:
    - hostvars[item].private_ip is defined

- name: "Build hosts file (using public network IPs)"
    ...
  when:
    - hostvars[item].public_ip is defined
    - hostvars[item].private_ip is not defined
```

Reply:
Will do it that way (and I agree with you, but that's what happens when you have a swiss knife) :)